indexer/src/main.ts

```diff
@@ -82,112 +82,126 @@ export default async function transform({
-      }
 
-      // Can be null if:
-      // 1. The event is part of the defined ignored events (see IGNORED_KEYS).
-      // 2. The event has an invalid number of keys.
-      const ethLogs = receipt.events
-        .map((e) => {
-          return toEthLog({
-            transaction: ethTx,
+      return { ...eventExtended, ethTx: ethTx };
+    })
+    // Can be null if:
+    // 1. The typed transaction if missing a signature param (v, r, s).
+    .filter((eventExtended) => eventExtended.ethTx)
+    .map((eventExtended) => {
+      const ethLogs = eventExtended.event.receipt.events
+        .map((e) =>
+          toEthLog({
+            transaction: eventExtended.ethTx,
             event: e,
             blockNumber,
             blockHash,
             isPendingBlock,
-          });
-        })
+          })
+        )
+        // Can be null if:
+        // 1. The event is part of the defined ignored events (see IGNORED_KEYS).
+        // 2. The event has an invalid number of keys.
         .filter((e) => e !== null) as JsonRpcLog[];
-      const ethLogsIndexed = ethLogs.map((log, index) => {
+
+      return {
+        ...eventExtended,
+        ethLogs,
+      };
+    })
+    .map((eventExtended) => {
+      const ethLogsIndexed = eventExtended.ethLogs.map((log, index) => {
         log.logIndex = index.toString();
         return log;
       });
 
       const ethReceipt = toEthReceipt({
-        transaction: ethTx,
+        transaction: eventExtended.ethTx,
         logs: ethLogsIndexed,
-        event,
+        event: eventExtended.event,
         cumulativeGasUsed,
         blockNumber,
         blockHash,
         isPendingBlock,
       });
 
-      cumulativeGasUsed += BigInt(ethReceipt.gasUsed);
-      // ethTx.transactionIndex can be null (if the block is pending) but
-      // Number(null) is 0 so this won't panic.
-      cumulativeGasUsages[Number(ethTx.transactionIndex)] = cumulativeGasUsed;
+      return {
+        ...eventExtended,
+        ethReceipt,
+        event: eventExtended.event,
+      };
+    });
 
-      // Add all the eth data to the store.
-      store.push({ collection: "transactions", data: { tx: ethTx } });
-      store.push({ collection: "receipts", data: { receipt: ethReceipt } });
-      ethLogs.forEach((ethLog) => {
-        store.push({ collection: "logs", data: { log: ethLog } });
-      });
+  filteredEvents.forEach((eventExtended) => {
+    cumulativeGasUsed += BigInt(eventExtended.ethReceipt.gasUsed);
+    // ethTx.transactionIndex can be null (if the block is pending) but
+    // Number(null) is 0 so this won't panic.
+    cumulativeGasUsages[Number(eventExtended.ethTx.transactionIndex)] =
+      cumulativeGasUsed;
 
-      // Add the logs bloom of the receipt to the block logs bloom.
-      const receiptLogsBloom = new Bloom(hexToBytes(ethReceipt.logsBloom));
-      blockLogsBloom.or(receiptLogsBloom);
+    // Add all the eth data to the store.
+    store.push({
+      collection: "transactions",
```

Comment:
Can we have collection name constants in our constants file and simply use these constants in order to avoid typo or confusions?